src/components/media-views.js

```diff
@@ -694,6 +694,23 @@ AFRAME.registerComponent("media-video", {
       if (url.startsWith("hubs://")) {
         const streamClientId = url.substring(7).split("/")[1]; // /clients/<client id>/video is only URL for now
         const stream = await NAF.connection.adapter.getMediaStream(streamClientId, "video");
+        // We subscribe to video stream notifications for this peer to update the video element
+        // This could happen in case there is an ICE failure that requires a transport recreation.
+        if (this._onStreamUpdated) {
+          NAF.connection.adapter.off("stream_updated", this._onStreamUpdated);
+        }
+        this._onStreamUpdated = async (peerId, kind) => {
+          if (peerId === streamClientId && kind === "video") {
+            // The video stream for this peer has been updated
+            const stream = await NAF.connection.adapter.getMediaStream(peerId, "video").catch(e => {
+              console.error(`Error getting video stream for ${peerId}`, e);
+            });
+            if (stream) {
+              videoEl.srcObject = new MediaStream(stream);
```

Comment:
Hi! I discovered this doesn't work unless you also call `videoEl.play()`, because if the track dies the video is stopped.

Reply:
Does this happen in all browsers? Can you provide STRs? I can can't really reproduce it with current master in FF/Chrome.

The `stream_update` event in **media-view.js** and **avatar-audio-source.js** should only be as a consequence of a _receive transport recreation_, otherwise both components are not  yet created so the `stream_update` event is not triggered on them. Maybe we are are missing some case where it's being triggered? I  haven't found any but having STRs would really help.

A _receive transport recreation_ should only happen in case where the transport has been closed but signaling is still opened. The transport disconnected/failed states are handled by the WebRTC stack and we should recover without any consumer recreation or transport updates whatsoever (so no `stream_update` events). 

I couldn't also reproduce the issue mentioned above. How are you triggering it?